rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -4,29 +4,35 @@ import cats.syntax.all._
 import cats.effect._
 import cats.effect.concurrent.{Deferred, Ref}
 import coop.rchain.rspace.history.HistoryReaderBase
-import coop.rchain.shared.MapOps._
 import coop.rchain.rspace.internal._
 
 import scala.collection.immutable.Map
 
 trait HotStore[F[_], C, P, A, K] {
-  def getContinuations(channels: Seq[C]): F[Seq[WaitingContinuation[P, K]]]
-  def putContinuation(channels: Seq[C], wc: WaitingContinuation[P, K]): F[Unit]
-  def installContinuation(channels: Seq[C], wc: WaitingContinuation[P, K]): F[Unit]
+  type Continuation    = WaitingContinuation[P, K]
+  type Continuations   = Seq[Continuation]
+  type Data            = Datum[A]
+  type HotStoreActions = Seq[HotStoreAction]
+  type RowMap          = Map[Seq[C], Row[P, A, K]]
+  type State           = HotStoreState[C, P, A, K]
```

Comment:
Type aliases are useful to make the code more readable when types are very long. But they must be used sparingly and only when really necessary because when reading the code you must have it in the head and recalculate.

So I'd leave `Seq[_]` and `Datum[_]` (or any pattern _[_]) as is because it's not complicated type.

`RowMap` is only used in trait definition and once in implementation, so it's not very beneficial to have it as alias and force reader to unpack it.

Because aliases are defined with trait type parameters (C, P, A, K) they cannot be used outside of trait scope, e.g. for `State` this means it cannot be used on many places which makes it not very helpful.

My intention for aliases was more towards this `Deferred[F, Seq[WaitingContinuation[P, K]]]` which is used multiple times in a scope of a function. But also the question is if it's necessary to use it when calling _modify_.

---

For alias to be justified it must satisfy these criteria:
- drastically reduce the length of type definition
- used in smallest possible scope or be a global.

These are not very precise rules but it can serve as guidelines. Aliases are great language feature but it can make code much harder to reason about if not used very carefully.

Reply:
@tgrospic, yes, I agree. Reasonable reasons. I tried to fix it.